src/esmock.d.ts

```diff
@@ -76,3 +101,4 @@ declare namespace esmock {
 }
 
 export default esmock;
+export { esmock as partial, esmock as strict };
```

Comment:
Probably if you're doing this, you need to not have the namespace above, maybe? Truthfully you're getting into an API that I can't reason about without having in TS first. Maybe try writing things into the TS playground and then read the d.ts tab.

(Honestly, this codebase isn't that big... It wouldn't be all that hard to convert it to TS)

Reply:
having both is necessary in order to support both of these
```javascript
import esmock, { partial } from 'esmock'

partial(...args) // from `export { esmock as partial }`
esmock.partial(...args) // from declare namespace esmock
```

Reply:
I suppose so, though I would write `export { esmock.partial as partial }`; that way you can write JSDoc comments that explain what each do individually.

Reply:
Oh, you want to write `esmock.partial.p` etc, which is recursive... I'm not sure if there are good steps to take to do that one. I'd personally recommend having a simpler API overall; if people want to do the dotted notation, they could just write `import * as esmock from 'esmock'`, right?

Reply:
for not-typescript users the current interface is good (imo)

Reply:
esm and typescipt don't allow this dynamic-ness `export { esmock.partial as partial }` it needs to export a single name and no lookups.

marking this as resolved as well